ddtrace/tracer/longrunner.go

```diff
@@ -0,0 +1,187 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"sync"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// TrackingExpirationLimit is the limit for how long to track a long-running span before no longer sending snapshots
+const TrackingExpirationLimit = 12 * time.Hour
+
+// Any span living longer than heartbeatInterval will have heartbeats sent every interval
+var heartbeatInterval time.Duration
+
+//TODO: is there a better performing design than this?
```

Comment:
IIRC, write-efficient concurrent maps try to avoid write locks by basically hashing the key into several distinct sub-maps - it's like having several "write lanes". Here is an intro to it: https://www.openmymind.net/Shard-Your-Hash-table-to-reduce-write-locks/ 

Do you think it could help? I didn't get the chance to look at the rest of the code to better understand this question but I felt it could already help.

Reply:
Thanks @Julio-Guerra, I was trying to say the same in my comment here https://github.com/DataDog/dd-trace-go/pull/1202#issuecomment-1062809744, but I think your description is more clear, and the article is also illustrating things well.

That being said, the implementation in the article looks pretty inefficient. Anyway, it's a good for clearly showing the main idea.